CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+project(libsimulator)
+cmake_minimum_required(VERSION 2.8.7)
+
+set(SRC_DIR src)
+set(TEST_SRC_DIR test)
+set(INCLUDE_DIR include)
+
+find_package(Boost COMPONENTS system REQUIRED)
+
+set(SIMULATOR_SRC_FILES
+  ${SRC_DIR}/acceptor.cpp
+  ${SRC_DIR}/high_resolution_timer.cpp
+  ${SRC_DIR}/io_service.cpp
+  ${SRC_DIR}/resolver.cpp
+  ${SRC_DIR}/sink_forwarder.cpp
+  ${SRC_DIR}/udp_socket.cpp
+  ${SRC_DIR}/default_config.cpp
+  ${SRC_DIR}/http_proxy.cpp
+  ${SRC_DIR}/pcap.cpp
+  ${SRC_DIR}/simulation.cpp
+  ${SRC_DIR}/socks_server.cpp
+  ${SRC_DIR}/high_resolution_clock.cpp
+  ${SRC_DIR}/http_server.cpp
+  ${SRC_DIR}/queue.cpp
+  ${SRC_DIR}/simulator.cpp
+  ${SRC_DIR}/tcp_socket.cpp
+)
+
+add_library(simulator ${SIMULATOR_SRC_FILES})
```

Comment:
This should work via [add_subdirectory](https://cmake.org/cmake/help/latest/command/add_subdirectory.html).

I don't think the include directories are propagated, but I'm not sure. The main reason I wanted to add cmake support is to make development on libsimulator itself easier (I find bjam confusing/limiting and some IDEs can easily auto-import the config from a `CMakeLists` file).